src/nix/command.cc

```diff
@@ -61,15 +66,48 @@ void StorePathsCommand::run(ref<Store> store)
     }
 
     else {
+        if (!recursive && (includeBuildDeps || includeEvalDeps))
+            throw UsageError("--build and --eval require --recursive");
```

Comment:
Would it be better to just assume `--recursive`?

Reply:
Perhaps, although I'm not sure how to handle `--no-recursive --build` or `--no-recursive --eval`. /That/ should be an error, but I don't think we have a way to tell whether that flag is set. Adding the assertion here is easier.

Reply:
There's no `--no-recursive` flag so I'm a bit confused :)

Reply:
`nix copy` will set recursive=true so that https://github.com/NixOS/nix/blob/c9d0cf7e02d4b3af3c027d7d74d5caa3c8963d26/src/nix/command.cc#L39 is active